Cabal/Distribution/Simple/Configure.hs

```diff
@@ -513,11 +554,19 @@ configure (pkg_descr0', pbi) cfg = do
     -- configuration.
     --
     -- TODO: Move this into a helper function.
-    defaultDirs :: InstallDirTemplates
+    defaultDirs0 :: InstallDirTemplates
         <- defaultInstallDirs (compilerFlavor comp)
                               (fromFlag (configUserInstall cfg))
                               (hasLibs pkg_descr)
-    let installDirs :: InstallDirTemplates
+    let defaultDirs
+            | use_external_internal_deps
+            -- Be a bit more hermetic about per-component installs
+            = defaultDirs0 {
+                datasubdir = toPathTemplate $ "$abi" </> "$libname",
+                docdir     = toPathTemplate $ "$datadir" </> "doc" </> "$abi" </> "$libname"
+              }
```

Comment:
Could we move these default values somewhere more obvious, e.g. next to where `defaultDirs` itself lives.